src/Control/Effect/Fresh.hs

```diff
@@ -18,10 +18,11 @@ module Control.Effect.Fresh
 ) where
 
 import Control.Algebra
+import Data.Kind (Type)
 
 -- | @since 0.1.0.0
-newtype Fresh m k = Fresh (Int -> m k)
-  deriving (Functor)
+data Fresh (m :: Type -> Type) k where
```

Comment:
I’ve gone out of my way to use `Type` instead of `*` for the new kind signatures. I’m planning on turning the star is type warning back on at some point this release.